### ISSUE TITLE
`make` errors out with random error on first run

### ISSUE THREAD

**Issue by Thermi** [markfasheh/ocfs2-tools#12]
Hello,

I'm trying to build version 1.8.5 as packaged in the release tarball for it on Arch Linux,
but it always errors out in the first run of the Makefile and succeeds in the second one.
The errors occur when I try to compile with -j1 or with the number of workers as the number of threads on the system.

`/usr/bin/python` is python3
`automake -v` 1.15
`gcc -v` .3.1
Shell is bash.
CFLAGS="-march=native -mtune=native -O2 -pipe -fstack-protector --param=ssp-buffer-size=4"
LDFLAGS="-Wl,-O1,--sort-common,--as-needed,-z,relro"

The errors can be one of the following:
 
```
make[1]: Entering directory '/home/thermi/aur/AUR4/ocfs2-tools/src/ocfs2-tools-ocfs2-tools-1.8.5/libo2cb'
gcc -march=native -mtune=native -O2 -pipe -fstack-protector --param=ssp-buffer-size=4 -g -fvar-tracking-assignments -Wall -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -pipe -fPIC  -D_FORTIFY_SOURCE=2  -I../include -I.   -MD -MP -MF ./.o2cb_abi.d -o o2cb_abi.o -c o2cb_abi.c
In file included from o2cb_abi.c:52:0:
../include/ocfs2/ocfs2.h:57:29: fatal error: ocfs2/ocfs2_err.h: No such file or directory
 #include <ocfs2/ocfs2_err.h>
                             ^
compilation terminated.
make[1]: *** [../Postamble.make:40: o2cb_abi.o] Error 1
```

```
make -C fsck.ocfs2
make[1]: Entering directory '/home/thermi/aur/AUR4/ocfs2-tools/src/ocfs2-tools-ocfs2-tools-1.8.5/fsck.ocfs2'
generating prompt-codes.h by parsing fsck.ocfs2.checks.8.in
make[1]: *** No rule to make target '../libo2cb/libo2cb.a', needed by 'fsck.ocfs2'.  Stop.
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/home/thermi/aur/AUR4/ocfs2-tools/src/ocfs2-tools-ocfs2-tools-1.8.5/fsck.ocfs2'
make: *** [Postamble.make:54: fsck.ocfs2] Error 2
```

**Comment by Thermi** [comment]
I think I figured it out. The Makefile structure does not specify the dependencies between the different software in the directories. Because some shared objects in some directories require other shared objects in some other directories to exist (because they link against them), the compilation fails when the directories are made out of order.

**Comment by josephhz** [comment]
Fixed by commit 269d3628b09e "Makefile: allow parallel builds".
